pkg/apis/machine/types.go

```diff
@@ -492,10 +496,27 @@ const (
 
 	// RollingUpdateMachineDeploymentStrategyType means that old MCs will be replaced by new one using rolling update i.e gradually scale down the old MCs and scale up the new one.
 	RollingUpdateMachineDeploymentStrategyType MachineDeploymentStrategyType = "RollingUpdate"
+
+	// InPlaceUpdateMachineDeploymentStrategyType signifies that machines will be updated in place,
+	// gradually transitioning from the old MachineSet to the new MachineSet without requiring machine recreation.
+	InPlaceUpdateMachineDeploymentStrategyType MachineDeploymentStrategyType = "InPlaceUpdate"
 )
 
 // RollingUpdateMachineDeployment specifies the spec to control the desired behavior of rolling update.
 type RollingUpdateMachineDeployment struct {
+	UpdateConfiguration
+}
+
+// InPlaceUpdateMachineDeployment specifies the spec to control the desired behavior of inplace rolling update.
+type InPlaceUpdateMachineDeployment struct {
+	UpdateConfiguration
+
+	// ManualUpdate when set to true, the user has to select the machines to be updated manually.
+	ManualUpdate bool
```

Comment:
Introduce an enum Type as suggested in https://github.com/gardener/gardener/pull/10828#discussion_r1928286418

Reply:
Adapted the API, PTAL.